cracl/ublox/msg/class/nav.cpp

```diff
@@ -577,6 +577,322 @@ bool timeutc::type(std::vector<uint8_t>& message)
+}
+
+int32_t timegps::fTOD()
+{
+  return m_fTOD;
+}
+
+int16_t timegps::week()
+{
+  return m_week;
+}
+
+int8_t timegps::leapS()
+{
+  return m_leapS;
+}
+
+uint8_t timegps::towValid()
+{
+  return m_towValid;
+}
+
+uint8_t timegps::weekValid()
+{
+  return m_weekValid;
+}
+
+uint8_t timegps::leapSValid()
+{
+  return m_leapSValid;
+}
+
+uint32_t timegps::tAcc()
+{
+  return m_tAcc;
+}
+
+bool timegps::type(std::vector<uint8_t>& message)
+{
+   return (!message.empty()
+      && valid_checksum(message)
+      && message[2] == ubx::msg_map.at("NAV").first
+      && message[3] == ubx::msg_map.at("NAV").second.at("TIMEGPS")); 
+}
+
+timegal::timegal(std::vector<uint8_t>& message)
+{
+  update(message);
+}
+
+void timegal::update(std::vector<uint8_t>& message)
+{
+  if (type(message))
+  {
+    m_iTOW = (*(reinterpret_cast<uint32_t*> (&message[6])));
+
+    m_galTow = (*(reinterpret_cast<uint32_t*> (&message[10])));
+
+    m_fGalTow = (*(reinterpret_cast<int32_t*> (&message[14])));
+
+    m_galWno = (*(reinterpret_cast<int16_t*> (&message[18])));
+
+    m_leapS = (*(reinterpret_cast<int8_t*> (&message[20])));
+
+    m_galTowValid = message[21] & 0x01;
+    m_galWnoValid = message[21] >> 1 & 0x01;
+    m_leapSValid = message[21] >> 2 & 0x01;
+
+    m_tAcc = (*(reinterpret_cast<uint32_t*> (&message[22])));
+  }
+  else
+    throw std::runtime_error("Message type mismatch");
+}
+
+uint32_t timegal::iTOW()
+{
+  return m_iTOW;
+}
+
+uint32_t timegal::galTow()
+{
+  return m_galTow;
+}
+
+int32_t timegal::fGalTow()
+{
+  return m_fGalTow;
+}
+
+int16_t timegal::galWno()
+{
+  return m_galWno;
+}
+
+int8_t timegal::leapS()
+{
+  return m_leapS;
+}
+
+uint8_t timegal::galTowValid()
+{
+  return m_galTowValid;
+}
+
+uint8_t timegal::galWnoValid()
+{
+  return m_galWnoValid;
+}
+
+uint8_t timegal::leapSValid()
+{
+  return m_leapSValid;
+}
+
+uint32_t timegal::tAcc()
+{
+  return m_tAcc;
+}
+
+bool timegal::type(std::vector<uint8_t>& message)
+{
+   return (!message.empty()
+      && valid_checksum(message)
+      && message[2] == ubx::msg_map.at("NAV").first
+      && message[3] == ubx::msg_map.at("NAV").second.at("TIMEGAL")); 
+}
+
+timebds::timebds(std::vector<uint8_t>& message)
+{
+  update(message);
+}
+
+void timebds::update(std::vector<uint8_t>& message)
+{
+  if (type(message))
+  {
+    m_iTOW = (*(reinterpret_cast<uint32_t*> (&message[6])));
+
+    m_SOW = (*(reinterpret_cast<uint32_t*> (&message[10])));
+
+    m_fSOW = (*(reinterpret_cast<int32_t*> (&message[14])));
+
+    m_week = (*(reinterpret_cast<int16_t*> (&message[18])));
+
+    m_leapS = (*(reinterpret_cast<int8_t*> (&message[20])));
+
+    m_sowValid = message[21] & 0x01;
+    m_weekValid = message[21] >> 1 & 0x01;
+    m_leapSValid = message[21] >> 2 & 0x01;
+
+    m_tAcc = (*(reinterpret_cast<uint32_t*> (&message[22])));
+  }
+  else
+    throw std::runtime_error("Message type mismatch");
+}
+
+uint32_t timebds::iTOW()
+{
+  return m_iTOW;
+}
+
+uint32_t timebds::SOW()
```

Comment:
same thing here with capitalization 
```suggestion
uint32_t timebds::sow()
```